.github/workflows/cilium-integration-tests.yaml

```diff
@@ -0,0 +1,187 @@
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Prepare variables for pushes to master
+        if: github.event_name == 'push'
+        run: |
+          echo "PROXY_IMAGE=quay.io/cilium/cilium-envoy" >> $GITHUB_ENV
+          echo "PROXY_TAG=${{ github.sha }}" >> $GITHUB_ENV
+
+      - name: Prepare variables for PR
+        if: github.event_name == 'pull_request'
+        run: |
+          echo "PROXY_IMAGE=quay.io/cilium/cilium-envoy-dev" >> $GITHUB_ENV
+          echo "PROXY_TAG=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+
+      - name: Prepare variables for issue comment on a PR
+        if: github.event_name == 'issue_comment'
+        run: |
+          echo "PROXY_IMAGE=quay.io/cilium/cilium-envoy-dev" >> $GITHUB_ENV
+          echo "PROXY_TAG=${{ github.event.issue.pull_request.head.sha }}" >> $GITHUB_ENV
+
+          commentBody="${{ github.event.comment.body }}"
+
+          ciliumRef=${CILIUM_REPO_REF}
+          if [[ "$commentBody" == *" cilium="* ]]; then
+            ciliumRef=$(echo "$commentBody" | sed -E 's|.* cilium=([^ ]*).*|\1|g')
+          fi
+          echo "CILIUM_REPO_REF=${ciliumRef}" >> $GITHUB_ENV
+
+          ciliumCliTag=${CILIUM_CLI_TAG}
+          if [[ "${{ github.event.comment.body }}" == *" ciliumCli="* ]]; then
+            ciliumCliTag=$(echo "$commentBody" | sed -E 's|.* ciliumCli=([^ ]*).*|\1|g')
+          fi
+          echo "CILIUM_CLI_TAG=${ciliumCliTag}" >> $GITHUB_ENV
+
+      - name: Reporting start to issue comment
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0
+        if: github.event_name == 'issue_comment'
+        with:
+          script: |
+            await github.rest.issues.updateComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              comment_id: ${{ github.event.comment.id }},
+              body: `${{ github.event.comment.body }}
+            
+              👋 https://github.com/${{ github.repository_owner }}/${{ github.event.repository.name }}/actions/runs/${{ github.run_id }}`
+            })
+            
+            await github.rest.reactions.createForIssueComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              comment_id: ${{ github.event.comment.id }},
+              content: 'rocket'
+            })
+
+      - name: Checkout Cilium ${{ env.CILIUM_REPO_REF }}
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          repository: cilium/cilium # Be aware that this is the Cilium repository and not the one of the proxy itself!
+          ref: ${{ env.CILIUM_REPO_REF }}
+
+      - name: Install Cilium CLI ${{ env.CILIUM_CLI_TAG }}
+        run: |
+          cid=$(docker create quay.io/cilium/cilium-cli-ci:${{ env.CILIUM_CLI_TAG }} ls)
```

Comment:
definitely. i switched from installing the Cilium CLI via `tar.gz` download from GitHub to the Docker Image approach because the `tar.gz` are only provided for releases whereas a Docker Image is built for every build.

But agree that it's not as comfortable to provide in case of a stable release. will address this together with the other mentioned improvements from you!

Reply:
addressed with PR https://github.com/cilium/proxy/pull/161